haddock-api/src/Haddock/Interface/Create.hs

```diff
@@ -767,7 +768,7 @@ availExportItem is_sig modMap thisMod semMod warnings exportedNames
               case M.lookup (nameModule t) instIfaceMap of
                 Nothing -> do
                    liftErrMsg $ tell
-                      ["Warning: Couldn't find .haddock for export " ++ pretty dflags t]
+                      [dislocatedErrMsg $ "Warning: Couldn't find .haddock for export " ++ pretty dflags t]
```

Comment:
As before, the `"Warning: "` is superfluous. Let's also pass the `moduleLocation` here.